### PR TITLE
Fix burn out lava river

### LINKED ISSUE #42598: throwing plasma onto lava makes land bridges
not sure what part of this interaction results in the bridge creation but:
1. mine lotsa plasma
2. throw like 10 sheets of ore onto lava
3. lava smelts ore into plasma sheets
4. lava ignites plasma sheets
5. plasma fire
6. plasma fire vanishes, bridge forms
7. escape the gulag

### PR DESCRIPTION
**Pull request by Dennok**
## About The Pull Request
Fix burn out lava river
fix#42598

## Why It's Good For The Game
Lava don't burn (mostly)

## Changelog
:cl:
fix: lava rivers no more burn out to basalt.
code: Now rivers generator can place a baseturf. 
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
I thought this was established as a feature.

**Comment by Dennok** [comment]
Code in lava freezing and use endothermic fusion to freezing the lavaland

**Comment by knacker48** [comment]
But this was such a cool interaction. It also gave a way for ashwalkers to cross lava rivers (that didn't involve using a RCD)

**Comment by Dennok** [comment]
But it have rcd on base

**Comment by necromanceranne** [comment]
It was more helpful for navigating lavaland without it being a pain in the ass. You did end up losing plasma ore but at least you don't have to endlessly backtrack finding your way across a lava river.

File: code/game/turfs/simulated/river.dm [hunk 1]
```diff
@@ -89,16 +89,17 @@
...
 
 	for(var/F in diagonal_turfs) //diagonal turfs only sometimes change, but will always spread if changed
 		var/turf/T = F
-		if(!istype(T, logged_turf_type) && prob(probability) && T.ChangeTurf(type, null, CHANGETURF_IGNORE_AIR))
+		if(!istype(T, logged_turf_type) && prob(probability) && T.ChangeTurf(type, list(type), CHANGETURF_IGNORE_AIR))
 			T.Spread(probability - prob_loss, prob_loss, whitelisted_area)
 		else if(ismineralturf(T))
 			var/turf/closed/mineral/M = T
-			M.ChangeTurf(M.turf_type, null, CHANGETURF_IGNORE_AIR)
+			M.ChangeTurf(M.turf_type, list(M.turf_type), CHANGETURF_IGNORE_AIR)
```
Comment (Dennok) [inline code/game/turfs/simulated/river.dm]: What this do? It change turf to himself? For what?

**Comment by YakumoChen** [comment]
That's a shame, it was really handy to get around lava rivers without needing to lug around extra tools.

**Comment by Qustinnus** [comment]
it's obviously not a feature. The tools for crossing lava exist, literaly just use those.

**Comment by Shadowflame909** [comment]
I have a very good reason for why this "bug" shouldn't be a bug and still needs to exist.

Literally yesterday I was playing and this happened





A literal perfect sea of lava surrounded mining everywhere but east.

This is some fucked up RNG and it'd be a shame if miners were required to get a firefighter/RCD and delay mats because RNG wasn't in their favor.

**Comment by Qustinnus** [comment]
sounds like you should report that as an issue seperate from this

**Comment by actioninja** [comment]
This is an exploit.

File: code/game/turfs/simulated/river.dm (hunk 1 above)
Comment (AnturK) [reply code/game/turfs/simulated/river.dm]: turf_type is not the same turf it's eg basalt floor for mineral walls.

**Pull request merged**